llvm/cmake/modules/LLVMConfig.cmake.in

```diff
@@ -52,6 +52,7 @@ set(LLVM_ENABLE_ASSERTIONS @LLVM_ENABLE_ASSERTIONS@)
 set(LLVM_ENABLE_EH @LLVM_ENABLE_EH@)
 
 set(LLVM_ENABLE_ONDISK_CAS @LLVM_ENABLE_ONDISK_CAS@)
+set(LLVM_CAS_ENABLE_REMOTE_CACHE @LLVM_CAS_ENABLE_REMOTE_CACHE@)
```

Comment:
Is there a way to avoid changing `llvm-config.h` based on whether `LLVM_CAS_ENABLE_REMOTE_CACHE` is enabled or not?

This header is affecting practically *every* compilation from LLVM, while whether remote cache is enabled or not should technically only affect a handful of places.

Reply:
But that is a build time configuration for whether the compile supports remote caching or not, not if remote caching is available or not. Are you expecting us to support both configuration in CAS CI?

The most useful place for this condition is actually unittest where we spin up different kinds of CAS to run on the same set of tests. I can change it to be based on runtime behavior but that will make breaking GRPC CAS undetectable.

Reply:
I mean you don't need to put a macro in `llvm-config.h`; if you have C++ code that needs to check how `LLVM_CAS_ENABLE_REMOTE_CACHE` was configured then you could pass a macro via the `CMakeLists.txt` of the target that contains such code.